client/my-sites/stats/stats-module/index.jsx

```diff
@@ -64,6 +64,25 @@ class StatsModule extends Component {
 		}
 	}
 
+	shouldComponentUpdate( nextProps ) {
+		for ( const key in nextProps ) {
+			if ( this.props[ key ] !== nextProps[ key ] ) {
+				if ( key !== 'data' ) {
+					return true; // found shallow unequal prop, should update.
+				}
+
+				// Special case for data prop, do deep equality check.
+				const isDeepEqual = isEqual( this.props[ key ], nextProps[ key ] );
```

Comment:
I've measured the time it takes for the whole `shouldComponentUpdate` to run, not just the `isEqual` and got 70ms for the entire interaction of transition from Reader to My Sites which reduce the wasted render from at least 600ms down to 40ms so it still worth it.